costs/anomalies/filters_route.go

```diff
@@ -0,0 +1,185 @@
+			db.RequestTransaction{Db: db.Db},
+			users.RequireAuthenticatedUser{users.ViewerCannot},
+			routes.RequestContentType{"application/json"},
+			routes.RequestBody{FiltersBody{
+				Filters: Filters{
+					Filter{
+						Rule: "product",
+						Data: []string{"NeededProduct1", "NeededProduct2"},
+					},
+				},
+			}},
+			routes.Documentation{
+				Summary:     "edit the anomalies filters",
+				Description: "Edits the anomalies filters based on the body",
+			},
+		),
+	}.H().Register("/costs/anomalies/filters")
+}
+
+// getAnomaliesFilters is a route handler which returns
+// the caller's list of filter.
+func getAnomaliesFilters(r *http.Request, a routes.Arguments) (int, interface{}) {
+	l := jsonlog.LoggerFromContextOrDefault(r.Context())
+	tx := a[db.Transaction].(*sql.Tx)
+	user := a[users.AuthenticatedUser].(users.User)
+	if dbUser, err := models.UserByID(tx, user.Id); err != nil {
+		l.Error("Failed to get user with id", map[string]interface{}{
+			"userId": user.Id,
+			"error":  err.Error(),
+		})
+		return http.StatusInternalServerError, errors.New("Failed to retrieve filters.")
+	} else {
+		filters := FiltersBody{Filters{}}
+		if dbUser.AnomaliesFilters != nil {
+			if err := json.Unmarshal(dbUser.AnomaliesFilters, &filters.Filters); err != nil {
+				l.Error("Failed to unmarshal anomalies filters", map[string]interface{}{
+					"userId": user.Id,
+					"error":  err.Error(),
+				})
+			}
+		}
+		return http.StatusOK, filters
+	}
+}
+
+// postAnomaliesFilters is a route handler which lets the user
+// add filters to their account.
+func postAnomaliesFilters(r *http.Request, a routes.Arguments) (int, interface{}) {
+	l := jsonlog.LoggerFromContextOrDefault(r.Context())
+	var body FiltersBody
+	routes.MustRequestBody(a, &body)
+	tx := a[db.Transaction].(*sql.Tx)
+	user := a[users.AuthenticatedUser].(users.User)
+	dbUser, err := models.UserByID(tx, user.Id)
+	if err != nil {
+		l.Error("Failed to get user with id", map[string]interface{}{
+			"userId": user.Id,
+			"error":  err.Error(),
+		})
+		return http.StatusInternalServerError, errors.New("Failed to update filters.")
+	}
+	return postAnomaliesFiltersWithValidBody(r, tx, dbUser, body)
+}
+
+// checkFilterValidity will check the filter name and type sent
+// in the body.
+func checkFilterValidity(filter Filter) bool {
+	for rule, dataType := range availableRules {
+		filterDataKind := reflect.ValueOf(filter.Data).Kind()
+		dataTypeKind := reflect.ValueOf(dataType).Kind()
+		if filter.Rule == rule && filterDataKind == dataTypeKind {
+			return true
+		}
+	}
+	return false
+}
+
+// postAnomaliesFiltersWithValidBody handles the logic assuming
+// the body is valid.
+func postAnomaliesFiltersWithValidBody(r *http.Request, tx *sql.Tx, dbUser *models.User, filters FiltersBody) (int, interface{}) {
+	for _, filter := range filters.Filters {
+		if !checkFilterValidity(filter) {
+			return http.StatusBadRequest, errors.New(filter.Rule + ": bad rule")
```

Comment:
Error response should be more specific (Rule name is invalid, rule data type is invalid, ...)